### ISSUE TITLE
Unexpected token < in JSON at position 0

### ISSUE THREAD

**Issue by Frogli** [SonarSource/atlasboard-sonarqube-package#7]
After I installed and configured the widget I get the error below when starting up atlasboard

```
17:16:26.93 <error> executed with errors: Unexpected token < in JSON at position 0 (in scheduler.js:37)
Call Stack:
    at handleError (/usr/local/lib/node_modules/atlasboard/lib/scheduler.js:37:29)
    at jobCallback (/usr/local/lib/node_modules/atlasboard/lib/scheduler.js:95:9)
    at <My Home folder>/dashboard/DevWallboard/packages/sonarqube/jobs/qg-filter/qg-filter.js:53:17
    at process._tickCallback (internal/process/next_tick.js:68:7)
```

Does anyone know what I am doing wrong and how to solve this?

Grtz

**Comment by erichirlemann** [comment]
`<` may be the start of an HTML tag.
Are you able tor reach the path `/api/projects/index` of your SQ instance?

**Comment by Frogli** [comment]
Thanks for your quick reply. Yes I can reach that path and get a nice json reponse. I had the wrong url which gave me some html output. Unfortunately I still can not get the widget to work. The error I get now is:
`15:10:53.42 <error> executed with errors: Bad status 400 (in Scheduler.js:21)
Call Stack:
    at handleError (/usr/local/lib/node_modules/status-board-cli/node_modules/status-board/lib/Scheduler.js:21:37)
    at jobCallback (/usr/local/lib/node_modules/status-board-cli/node_modules/status-board/lib/Scheduler.js:62:21)
    at /Users/h.p.noppe/Data/LocalRepositories/OmniHis Repos/dashboard/DevWallboard/packages/sonarqube/jobs/qg-filter/qg-filter.js:53:17
    at process._tickCallback (internal/process/next_tick.js:68:7)`

The root url I use in the global config is: http://localhost:9000/api/components/search?qualifiers=TRK

The user I use for this widget is a user which does not have System Privileges. I googled around a bit to get around this and found the url mentioned above. 

globalconfig
`
{
	"sonarqube": {
		"rootURL": "http://localhost:9000/api/components/search?qualifiers=TRK",
		"username": "user",
		"password": "P@ssw0rd"
	}
}
`

widget config
`
"qg-filter": {
      "interval": 180000,
      "widgetTitle": "Quality Gate Failures",
      "days_since_last_commitMax": 30,
      "credentials": "sonarqube"
    }
`

**Comment by erichirlemann** [comment]
According to https://github.com/SonarQubeCommunity/atlasboard-sonarqube-package/blob/80b8d704989a2f3661feeec2c7987f77ae06f6f9/jobs/qg-filter/qg-filter.js#L78, the job compose the url assuming rootURL looks like: `http://localhost:9000`
Would you like to try with this last value instead?

**Comment by Frogli** [comment]
ah you are right. I do not need it. I changed the rootURL to `http://localhost:9000`. Still I get the same error. I tested the url http://localhost:9000/api/projects/index?format=json with the same credentials as basis authentication with Postman and I get a nice json repsonse.

**Comment by erichirlemann** [comment]
True that the job does not show any relevant error message.
You have to guess what's wrong 🤔.
 What version of SQ do you run?

**Comment by Frogli** [comment]
He thanks for your quick reply. 

> What version of SQ do you run?

Oh sorry I should have mentioned that in the first place. I use SonarQube Version 7.4 (build 18908).

I dug a bit deeper in the code of the job since I did not get any more useful errors messages. And discovered that the problem is probably the next http call per project. 
`http://localhost:9000/api/measures/component?baseComponentKey=nl.omnihis.scipio:dex-sms&metricKeys=quality_gate_details,alert_status,last_commit_date`

This gives the following json response:
```JSON
{
    "errors": [
        {
            "msg": "Metrics quality_gate_details can't be requested in this web service. Please use api/measures/component"
        }
    ]
}
```

Then I changed the url to `/api/measures/component?componentKey=MyProjectKey&metricKeys=quality_gate_details,alert_status,last_commit_date`. Notice that I also had to change `baseComponentKey' to `componentKey`. This gave me a nice json output again. But the code needed some changes. 

I forked the repository, made the changes and created a [pull request](https://github.com/SonarQubeCommunity/atlasboard-sonarqube-package/pull/8).
